src/targets/gpu/write_literals.cpp

```diff
@@ -27,36 +27,155 @@
 #include <migraphx/instruction.hpp>
 #include <migraphx/program.hpp>
 #include <migraphx/env.hpp>
+#include <migraphx/liveness.hpp>
+#include <migraphx/register_op.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_COPY_LITERALS)
 
+struct literal_as_argument
+{
+    std::string name() const { return "gpu::literal_as_argument"; }
+    argument data;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.data, "data"));
+    }
+
+    shape compute_shape(const std::vector<shape>&) const { return data.get_shape(); }
+    argument compute(context&, const shape&, const std::vector<argument>&) const { return data; }
+    friend std::ostream& operator<<(std::ostream& os, const literal_as_argument& x)
+    {
+        os << x.name() << "[data=";
+        if(x.compute_shape(std::vector<shape>{}).elements() < 10)
+        {
+            os << x.data;
+        }
+        else
+        {
+            os << "{ ... }";
+        }
+        os << "]";
+        return os;
+    }
+};
+MIGRAPHX_REGISTER_OP(literal_as_argument);
+
 void write_literals::apply(module& m) const
 {
     assert(ctx != nullptr);
     std::size_t n = 0;
-    for(auto ins : iterator_for(m))
+
+    if(weight_streaming)
     {
-        if(ins->name() == "@literal")
+        std::size_t bytes_on_gpu = 0;
```

Comment:
This needs to be set to the size of the scratch memor+total bytes of all parameters.

Reply:
Changed, thanks